test.py

```diff
@@ -251,6 +251,33 @@ def test_mixed_case_checksums(self):
         bag = bagit.Bag(self.tmpdir)
         self.assertTrue(self.validate(bag))
 
+    def test_unsafe_directory_entries_raise_error(self):
+        bad_paths = None
+        # This could be more granular, but ought to be
+        # adequate.
+        if os.name == 'nt':
+            bad_paths = (
+                r'C:\win32\cmd.exe',
+                '\\\\?\\C:\\',
+                'COM1:',
+                '\\\\.\\COM56'
+            )
+        else:
+            bad_paths = (
+                '../../../secrets.json',
+                '~/.pgp/id_rsa',
+                '/dev/null'
```

Comment:
The other one I'd add – and I think the `..` tests should be cross-platform – would be something which starts relative and uses `..` to escape like `data/../../../../../etc/passwd`.